clippy_utils/src/sugg.rs

```diff
@@ -394,7 +394,34 @@ impl Neg for Sugg<'_> {
 impl Not for Sugg<'_> {
     type Output = Sugg<'static>;
     fn not(self) -> Sugg<'static> {
-        make_unop("!", self)
+        use AssocOp::{Equal, Greater, GreaterEqual, Less, LessEqual, NotEqual};
+
+        /// Convert ```AssocOp``` to a string of operators.
+        fn op_as_str(op: AssocOp) -> &'static str {
+            op.to_ast_binop().unwrap().to_string()
+        }
+
+        /// Replace the operator in the Snippet.
+        fn replace_op(from_op: AssocOp, to_op: AssocOp, snip: Cow<'_, str>) -> Sugg<'static> {
+            let from = op_as_str(from_op);
+            let to = op_as_str(to_op);
+            let snip = snip.into_owned().replace(from, to);
```

Comment:
I think we should avoid using string replacement since it could replace the wrong thing. Could we refactor `Sugg::BinOp` to contains two snippets for left and right side?

Reply:
Could be good. I'll give it a try.

Reply:
This is how it turned out for now. What do you think?

Reply:
Can we do without the snippet of the entire expression?

Reply:
That's better. Thanks!